### PR TITLE
Update Julia queries

### PR DESCRIPTION
**Pull request by savq**
* Update queries for `@parameter_outer` in:
    - call_expression / argument_list
    - function_definition / parameter_list
* Add queries for:
    - structs
    - math functions
    - macros

### CODE REVIEW

File: queries/julia/textobjects.scm [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+(struct_definition) @class.outer    ; not classes, but close enough
 
 (comment) @comment.outer
 
 (if_statement) @conditional.outer
 
-(function_definition) @function.outer
-(function_expression) @function.outer    ;; lambdas
+(function_definition
+  name: (identifier)
```
Comment (theHamsta) [inline queries/julia/textobjects.scm:15]: why do we need the `name` in the query? Can't the function as well be anonymous and if not then `name: (identifier)` does not really add a benefit.-

**Review by theHamsta — APPROVED** [review]
Nice! Really glad to receive this addition!

File: queries/julia/textobjects.scm (hunk 1 above)
Comment (theHamsta) [reply queries/julia/textobjects.scm:15]: Ok. It's a function_expression without the name.

Comment (savq) [reply queries/julia/textobjects.scm:15]: I actually forgot to check that case 😬

Good thing it's covered by the other query, although it does seem a bit confusing. Not sure why the grammar would be defined that way.

**Pull request merged**